### ISSUE TITLE
Support requesting code-changes

### ISSUE THREAD

**Issue by rlch** [ldelossa/gh.nvim#5]
Hey!

First of all, loving the look of this plugin - I'm actively watching its development. Haven't used it yet as I'll be waiting for the all commits flag; which is more inline with my workflow.

I was wondering if you support / have plans on supporting requesting code-changes? One of the biggest frustration points for me when reviewing PR's is not having LSP + treesitter support in the GitHub UI, resulting in requested changes with typo's + indentation issues etc. Would love to get your thoughts on this!

Thanks :)

**Comment by rlch** [comment]
![image](https://user-images.githubusercontent.com/35163478/168053439-a60c6ed9-c759-4be9-a7fb-f58f62ffebd2.png)

![image](https://user-images.githubusercontent.com/35163478/168054076-ca442a89-e555-4562-a3b8-bb892fc36938.png)

Sorry @ldelossa, should've clarified. It's actually called a suggestion

**Comment by ldelossa** [comment]
> First of all, loving the look of this plugin - I'm actively watching its development. Haven't used it yet as I'll be waiting for the all commits flag; which is more inline with my workflow.

By this do you mean you're waiting for a tree which shows "all modified files" ?

**Comment by rlch** [comment]
> > First of all, loving the look of this plugin - I'm actively watching its development. Haven't used it yet as I'll be waiting for the all commits flag; which is more inline with my workflow.
> 
> By this do you mean you're waiting for a tree which shows "all modified files" ?

Yep. In fact, my workflow is a bit of a mix of the two depending on the context/size of the PR. So ideally it's not just a flag, I can select whether to review all modified files or each commit through the UI

**Comment by ldelossa** [comment]
Cool suggestions! Let me look into a few things. I need to do a couple other PRs until we get to suggestions (like aggregate file view we mentioned above). I also need to brush up on TS. I dont really know if I can apply TS highlighting to a chunk of arbitrary text inside a buffer? I think I see nvim-notify doing this, but not sure how yet. 

In a perfect world, any code blocks in the covo buffer of a threaded message get a nice syntax aware highlighting. And yes, morphing "suggestion" markdown blocks into inline diffs would be pretty sick.

**Comment by joestringer** [comment]
For posting code suggestions, my workflow currently is to open the change that I want to comment on, yank the relevant lines, open a comment thread, then manually type ```` ```suggestion ````, newline, paste, edit , type the ```` ``` ```` to close out the diff, and post the comment.

It would be neat to have something a bit more integrated like if you edit/delete some lines in the file buffer then open a comment thread, the buffer automatically opens with a ```` ```suggestion ... ``` ```` comment pre-populated.

Extra points would be the inverse workflow - when reading reviews from peers, ability to integrate the suggestions directly back into the local files. I don't know quite how that would look though.
